server/rest/src/main/java/org/infinispan/rest/server/InfinispanResponse.java

```diff
@@ -0,0 +1,153 @@
+package org.infinispan.rest.server;
+
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.Date;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+import org.infinispan.rest.server.operations.mediatypes.MediaType;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufUtil;
+import io.netty.buffer.Unpooled;
+import io.netty.handler.codec.http.DefaultFullHttpResponse;
+import io.netty.handler.codec.http.FullHttpResponse;
+import io.netty.handler.codec.http.HttpHeaderNames;
+import io.netty.handler.codec.http.HttpHeaderValues;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.netty.handler.codec.http.HttpUtil;
+import io.netty.handler.codec.http.HttpVersion;
+import io.netty.handler.codec.http2.HttpConversionUtil;
+
+public class InfinispanResponse {
```

Comment:
Well I wanted the name of this class to stay away from common names used by HTTP frameworks. Currently on REST Server module there are 16 `Http*Builder` classes from Netty and HTTP Components. Let's stay away from that sort of names. But of course I'm open for any other alternatives.

Those `Optional` fields are very convenient when creating `FullHttpResponse` (see `#toNettyHttpResponse`). I prefer to keep it clean rather than doing non-null checks all around.

BTW, I plan to create a JMH test before replacing existing solution. If I spot any performance problem there, I would be happy to replace those Optional fields.